### PR TITLE
download script for ensembl ftp

### PR DESCRIPTION
**Pull request by bernt-matthias**
basically the same as the other PR. checksums are not md5 here. maybe you have a nicer idea to use `sum`?

### CODE REVIEW

**Comment by bernt-matthias** [comment]
hab 2 travis checks hinzugefuegt (ensembl hat 2 verschiedene checksum files .. und manchmal keins). PR scheint mir fertig.

**Comment by wookietreiber** [comment]
Please stick to English.

Please rebase. I have just added a [contribution guide](https://github.com/idiv-biodiversity/scddl/blob/master/CONTRIBUTING.md) for reference (contains the command line instructions for a rebase).

File: ensembldl.sh [hunk 1]
```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env bash
```
Comment (wookietreiber) [inline ensembldl.sh]: This file has execute permissions. Unfortunately, GitHub does not show this, so I'm commenting on the shebang. Please `chmod -x ensembldl.sh`.

File: ensembldl.sh [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+# -----------------------------------------------------------------------------
+# usage
+# -----------------------------------------------------------------------------
+
+function usage { cat << EOF
+$app $version
+
+USAGE
+
+  $app [options] [--] dataset output
```
Comment (wookietreiber) [inline ensembldl.sh]: Please change this to `$app [options] [--] prefix dataset...` to be consistent with the latest changes in `ncbidl.sh`.

File: .travis.yml [hunk 3]
```diff
@@ -22,6 +22,8 @@ before_script:
 
 script:
   - bash ncbidl.sh -v -p 2 /tmp/db blast/db/vector
+  - bash ensembldl.sh -v -p 2 pub/release-93/fasta/gallus_gallus/dna/Gallus_gallus.Gallus_gallus-5.0.dna.chromosome.3 /tmp/db
+  - bash ensembldl.sh -v -p 2 pub/release-93/bed/ensembl-compara/70_mammals.gerp_constrained_element/gerp_constrained_elements.bos_taurus /tmp/db
```
Comment (wookietreiber) [inline .travis.yml]: Please split the command(s) into multiple lines using `- >-` as is done with the `diamonddb.sh` test. This improves code readability.

File: ensembldl.sh (hunk 1 above)
Comment (bernt-matthias) [reply ensembldl.sh]: done

File: ensembldl.sh (hunk 2 above)
Comment (bernt-matthias) [reply ensembldl.sh]: done. incorporated all the changes from ncbidl.

File: .travis.yml (hunk 3 above)
Comment (bernt-matthias) [reply .travis.yml]: done

**Comment by bernt-matthias** [comment]
I rebased, but could not push. 

```
error: failed to push some refs to 'https://github.com/bernt-matthias/scddl.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

**Comment by bernt-matthias** [comment]
has been force pushed

File: ensembldl.sh [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+function usage { cat << EOF
+$app $version
+
+USAGE
+
+  $app [options] [--] prefix dataset...
+
+DESCRIPTION
+
+  download NCBI data set
```
Comment (wookietreiber) [inline ensembldl.sh]: This shouldn't be **NCBI** anymore, should it?

File: ensembldl.sh [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+EOF
+}
+
+# -----------------------------------------------------------------------------
+# application
+# -----------------------------------------------------------------------------
+
+for dataset in "${datasets[@]}"
+do
+  output_basedir="$prefix/ncbi/$dataset"
```
Comment (wookietreiber) [inline ensembldl.sh]: This should also be changed to **ensembl** instead of **ncbi**, right?

Comment (bernt-matthias) [reply ensembldl.sh]: done

File: ensembldl.sh (hunk 4 above)
Comment (bernt-matthias) [reply ensembldl.sh]: done

**Comment by wookietreiber** [comment]
I have enabled all the checks Codacy provides yesterday. Please fix these. If you want to test this locally, you can use the **shellcheck** tool: `shellcheck *.sh`.

You can ignore the **Not following: util.sh** error as is reported by Codacy. They seem to be doing something wrong there, most likely checking each file individually instead of using globbing and not providing the `-x` argument to shellcheck. I'll let them know about this.

**Comment by wookietreiber** [comment]
Something is going wrong with the tests on Travis. It looked a bit like sporadic, so I restarted, same error, though. Please investigate.

**Comment by wookietreiber** [comment]
Do you have an example ftp directory for when there are neither `MD5SUM` nor `CHECKSUMS` files?

**Comment by bernt-matthias** [comment]
Yep, I do. Added it to .travis.yml

**Comment by wookietreiber** [comment]
Thanks! I squashed, so it's in master now :)

**Pull request closed without merging**